src/Plugin/EPoetryLogPlugin.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Plugin;
+
+use Phpro\SoapClient\Event\FaultEvent;
+use Phpro\SoapClient\Event\RequestEvent;
+use Phpro\SoapClient\Event\ResponseEvent;
+use Phpro\SoapClient\Events;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class ePoetryLogPlugin
+ */
+class EPoetryLogPlugin implements EventSubscriberInterface
+{
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var bool | string
+     */
+    protected $logLevel = false;
+
+    /**
+     * Constructor
+     *
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger, $log_level)
+    {
+        $this->logger = $logger;
+        $this->logLevel = $log_level;
+    }
+
+    /**
+     * @param RequestEvent $event
+     */
+    public function onClientRequest(RequestEvent $event): void
+    {
+        $this->logInfo(sprintf(
+            '[phpro/soap-client] request: call "%s" with params %s',
```

Comment:
We can use here `[epoetry] ...` or `ePoetry: ....` instead of `[phpro/soap-client] ...`.